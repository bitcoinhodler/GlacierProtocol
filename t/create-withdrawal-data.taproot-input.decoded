{
  "txid": "362cd0baa47c4110b2338c5bb1515d0e88fc52c4caf2b4d643211d252492a127",
  "hash": "e5c93c0b1498cfc1b9ccf6827fdfc07f6d4966af738a5c985ef39f1136cf03d9",
  "version": 2,
  "size": 382,
  "vsize": 166,
  "weight": 664,
  "locktime": 0,
  "vin": [
    {
      "txid": "a7337a2850a4425df3e4fe7e4f86d7f2403cf11179657ade795027593e466c8e",
      "vout": 1,
      "scriptSig": {
        "asm": "",
        "hex": ""
      },
      "txinwitness": [
        "",
        "30440220671e76c42685bfe60fff09ebb8a6ac98ed2bbb65b3bfcec747be665a7005254c02201297fcf91b3a806b19eb992034418c416306868886a5716aae9a273cd371101701",
        "304402201000e16d75f896688ce4e2e16ad798aa05f2890ab279f10686647f885d66c2a90220444c7d9199b54e8af91b3f66a3330ba25c09425daf7d7a0e9078b980222641c501",
        "522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae"
      ],
      "sequence": 4294967295
    }
  ],
  "vout": [
    {
      "value": 0.00009834,
      "n": 0,
      "scriptPubKey": {
        "asm": "1 901069f8e009bb67a27b47054150c60287b0af5511bb214facfebe0945114df4",
        "desc": "rawtr(901069f8e009bb67a27b47054150c60287b0af5511bb214facfebe0945114df4)#4chsgtwg",
        "hex": "5120901069f8e009bb67a27b47054150c60287b0af5511bb214facfebe0945114df4",
        "address": "bcrt1pjqgxn78qpxak0gnmguz5z5xxq2rmpt64zxajznavl6lqj3g3fh6qlvaknf",
        "type": "witness_v1_taproot"
      }
    }
  ]
}
