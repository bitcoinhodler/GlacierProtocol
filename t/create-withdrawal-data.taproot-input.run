#!/bin/bash
set -e

# Test withdrawing from transaction that also has a taproot output.
# We send our withdrawal to a taproot output too.

$GLACIERSCRIPT --testnet=$1 create-withdrawal-data << INPUT 2>&1
y
y
y
y
y
y
tb1qvhyx836sxvwvq0hmfkk2y9l8jfw0lpl4a89fjf6kgg6turtz0r5q8hz22g
522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae
tb1pjqgxn78qpxak0gnmguz5z5xxq2rmpt64zxajznavl6lqj3g3fh6qj4hsxn
1
02000000000101432d55803a4295ed6af9021f21fef8c6e1774f50d5db1b8fcbe380ef12e7a20e0100000000fdffffff02f0270000000000002251201662b0217bba4013e02d7ff9339ac1996cf795a40bb5722ee57d844127e60fe3102700000000000022002065c863c750331cc03efb4daca217e7925cff87f5e9ca9927564234be0d6278e802473044022075d3bd6b3ed03434ccd08e3f9317ccac429f4a457e38b06b11cee7a0f306355b022071651c76784e76cd9b6762946156deadc5a4a2c726df5db5a3693b852255bfd70121031c7c09fc11765146a35a0bf6861f5e630cf231f4640e46d80b25f5562a345bd000000000
2
cNYaH3onqrdMffpznhMMmrHn34fuTU59w5j8LM3H42VPcUsLeXy5
cRoydfinDRzzRQJp5niqJWukSYTfPJQM6ytqGN6nzonaz1mafgwD
1
y

y
INPUT


# Cleanup cruft
rm transaction.png
