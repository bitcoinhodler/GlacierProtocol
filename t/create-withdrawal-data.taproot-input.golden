Are you running this on a computer WITHOUT a network connection of any kind? (y/n)?Have the wireless cards in this computer been physically removed? (y/n)?Are you running on battery power? (y/n)?Are you running on an operating system booted from a USB drive? (y/n)?Is your screen hidden from view of windows, cameras, and other people? (y/n)?Are smartphones and all other nearby devices turned off and in a Faraday bag? (y/n)?
You will need to enter several pieces of information to create a withdrawal transaction.


*** PLEASE BE SURE TO ENTER THE CORRECT DESTINATION ADDRESS ***


Source cold storage address: 
Redemption script for source cold storage address: 
Destination address: 
How many unspent transactions will you be using for this withdrawal? 
Please paste raw transaction #1 (hexadecimal format) with unspent outputs at the source address
OR
input a filename located in the current directory which contains the raw transaction data
(If the transaction data is over ~4000 characters long, you _must_ use a file.):

Transaction data found for source address.
TOTAL unspent amount for this raw transaction: 0.00010000 BTC

How many private keys will you be signing this transaction with (at least 2 required to complete transaction)?
#: Key #1: Key #2: 
Enter fee rate.
Satoshis per vbyte: 
Based on the provided rate, the fee will be 0.00000209 bitcoin.
Confirm? (y/n): 
Please enter the decimal amount (in bitcoin) to withdraw to the destination address.

Example: For 2.3 bitcoins, enter "2.3".

After a fee of 0.00000209, you have 0.00009791 bitcoins available to withdraw.

*** Technical note for experienced Bitcoin users:  If the withdrawal amount & fee are cumulatively less than the total amount of the unspent transactions, the remainder will be sent back to the same cold storage address as change. ***

Amount to send to tb1pjqgxn78qpxak0gnmguz5z5xxq2rmpt64zxajznavl6lqj3g3fh6qj4hsxn (leave blank to withdraw all funds stored in these unspent transactions): With no change output, the transaction fee is reduced, and 0.00009834 BTC will be sent to your destination.

Is this data correct?
*** WARNING: Incorrect data may lead to loss of funds ***

0.00010000 BTC in unspent inputs from cold storage address tb1qvhyx836sxvwvq0hmfkk2y9l8jfw0lpl4a89fjf6kgg6turtz0r5q8hz22g
0.00009834 BTC going to destination address tb1pjqgxn78qpxak0gnmguz5z5xxq2rmpt64zxajznavl6lqj3g3fh6qj4hsxn
Fee amount: 0.00000166

Signing with private keys: 
cNYaH3onqrdMffpznhMMmrHn34fuTU59w5j8LM3H42VPcUsLeXy5
cRoydfinDRzzRQJp5niqJWukSYTfPJQM6ytqGN6nzonaz1mafgwD


Confirm? (y/n): 
Calculating transaction...

Final fee rate: 1.00 satoshis per vbyte

Raw signed transaction (hex):
020000000001018e6c463e59275079de7a657911f13c40f2d7864f7efee4f35d42a450287a33a70100000000ffffffff016a26000000000000225120901069f8e009bb67a27b47054150c60287b0af5511bb214facfebe0945114df404004730440220671e76c42685bfe60fff09ebb8a6ac98ed2bbb65b3bfcec747be665a7005254c02201297fcf91b3a806b19eb992034418c416306868886a5716aae9a273cd37110170147304402201000e16d75f896688ce4e2e16ad798aa05f2890ab279f10686647f885d66c2a90220444c7d9199b54e8af91b3f66a3330ba25c09425daf7d7a0e9078b980222641c5018b522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae00000000

Transaction fingerprint (md5):
f07417dc2b024ce366df6e24a8845dfa
QR code for transaction written to transaction.png
